ports/stm32/eth.c

```diff
@@ -381,6 +371,14 @@ STATIC int eth_mac_init(eth_t *self) {
                 break;
             case 1:
                 if (bsr & PHY_BSR_LINK_STATUS) {
+                    // Announce all modes
+                    eth_phy_write(self->phy_addr, PHY_ANAR,
+                        PHY_ANAR_SPEED_10HALF |
+                        PHY_ANAR_SPEED_10FULL |
+                        PHY_ANAR_SPEED_100HALF |
+                        PHY_ANAR_SPEED_100FULL |
+                        PHY_ANAR_IEEE802_3);
```

Comment:
This new setting is a change from the previous behaviour, which just left this register in its default state.  How does the change you've made impact things?  (It looks like a good change, I'm just wondering why you made it.)

Reply:
I made it to be sure that all possible modes that are supported by the Ethernet controller are announced. The default mode can as well be set by strapping pins at the PHY, which by chance/bad luck may be more restrictive.